rmd_files/strings.Rmd

```diff
@@ -272,10 +272,34 @@ stringr::str_replace_all(colours, "[^[A-Za-z0-9]]", "_")
 
 In the first example only the first match in each string has been replaced with an underscore, whereas in the second example all matches have been replaced.
 
+
 ---
 
 ### Exercise 1
 
+The column names of a table in an earlier exercise are: `offence_code`, `count_2016`, `count_2017`, `count_2018`, `count_2019`, `count_2020`.
+
+
+Using only functions covered in this part of the course, find the most efficient way to code this from scratch as:
+a) A vector of strings
+b) A single string, with column names separated by a comma and a space
+
+**Hint:** You may want to create an object in the first part of the question, and recycle it for the second part.
+
+---
+
+### Exercise 2
+
+The 'billboard' data set from the `tidyr` package contains US weekly music sales data.
+
+Extract its column names with `colnames(tidyr::billboard)`, and return an integer corresponding to the number of columns that contain `wk` ('week') in their names.
+
+**Hint:** Remember that booleans have numeric value (`TRUE == 1`, `FALSE == 0`)
```

Comment:
Is it worth directing participants towards the `str_detect()` function in the hint? Just because we've introduced quite a few new functions in this chapter, so it could take them a while to figure out which one they need if they're new to programming

Reply:
yeah!